resources/user.go

```diff
@@ -0,0 +1,298 @@
+	recWatcher *recwatch.RecWatcher
+}
+
+// Default returns some sensible defaults for this resource.
+func (obj *UserRes) Default() Res {
+	return &UserRes{
+		BaseRes: BaseRes{
+			MetaParams: DefaultMetaParams, // force a default
+		},
+	}
+}
+
+// Validate if the params passed in are valid data.
+func (obj *UserRes) Validate() error {
+	if obj.State != "exists" && obj.State != "absent" {
+		return fmt.Errorf("State must be 'exists' or 'absent'")
+	}
+	return obj.BaseRes.Validate()
+}
+
+// Init initializes the resource.
+func (obj *UserRes) Init() error {
+	return obj.BaseRes.Init() // call base init, b/c we're overriding
+}
+
+// Watch is the primary listener for this resource and it outputs events.
+func (obj *UserRes) Watch() error {
+	var err error
+	obj.recWatcher, err = recwatch.NewRecWatcher(passwdFile, false)
+	if err != nil {
+		return err
+	}
+	defer obj.recWatcher.Close()
+
+	// notify engine that we're running
+	if err := obj.Running(); err != nil {
+		return err // bubble up a NACK...
+	}
+
+	var send = false // send event?
+	var exit *error
+
+	for {
+		if obj.debug {
+			log.Printf("Watching: %s", passwdFile) // attempting to watch...
+		}
+
+		select {
+		case event, ok := <-obj.recWatcher.Events():
+			if !ok { // channel shutdown
+				return nil
+			}
+			if err := event.Error; err != nil {
+				return errwrap.Wrapf(err, "Unknown %s watcher error", obj)
+			}
+			if obj.debug { // don't access event.Body if event.Error isn't nil
+				log.Printf("%s: Event(%s): %v", obj, event.Body.Name, event.Body.Op)
+			}
+			send = true
+			obj.StateOK(false) // dirty
+
+		case event := <-obj.Events():
+			if exit, send = obj.ReadEvent(event); exit != nil {
+				return *exit // exit
+			}
+			//obj.StateOK(false) // dirty // these events don't invalidate state
+		}
+
+		// do all our event sending all together to avoid duplicate msgs
+		if send {
+			send = false
+			obj.Event()
+		}
+	}
+}
+
+// CheckApply method for User resource. Does nothing, returns happy!
+func (obj *UserRes) CheckApply(apply bool) (checkOK bool, err error) {
+	log.Printf("%s: CheckApply(%t)", obj, apply)
+
+	var exists = true
+	usr, err := user.Lookup(obj.GetName())
+	if err != nil {
+		if err == err.(user.UnknownUserError) {
+			log.Printf("the user: %s does not exist", obj.GetName())
+			exists = false
+		} else {
+			return false, errwrap.Wrapf(err, "error looking up user")
+		}
+	}
+
+	if obj.AllowDuplicateUID == false && obj.UID != nil {
+		existingUID, err := user.LookupId(strconv.Itoa(int(*obj.UID)))
+		if err == nil && existingUID.Username != obj.GetName() {
+			return false, fmt.Errorf("the uid: %d is already taken", *obj.UID)
+		}
+	}
+
+	if exists && obj.State == "exists" {
+		usercheck := true
+		if obj.GID != nil {
+			if strconv.Itoa(int(*obj.GID)) != usr.Gid {
+				usercheck = false
+			}
+		}
+		if obj.UID != nil {
+			if strconv.Itoa(int(*obj.UID)) != usr.Uid {
+				usercheck = false
+			}
+		}
+		if obj.HomeDir != "" && obj.HomeDir != usr.HomeDir {
```

Comment:
I just noticed that we might have three states:

1) HomeDir is some path "/home/james"
2) HomeDir is empty ""
3) Don't touch HomeDir (home dir is nil)

So I think this should be a *string... Do you agree?